flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemTableSink.java

```diff
@@ -284,4 +337,125 @@ public boolean configurePartitionGrouping(boolean supportsGrouping) {
 		this.dynamicGrouping = supportsGrouping;
 		return dynamicGrouping;
 	}
+
+	/**
+	 * Table bucket assigner, wrap {@link PartitionComputer}.
+	 */
+	private static class TableBucketAssigner implements BucketAssigner<BaseRow, String> {
+
+		private final PartitionComputer<BaseRow> computer;
+
+		private TableBucketAssigner(PartitionComputer<BaseRow> computer) {
+			this.computer = computer;
+		}
+
+		@Override
+		public String getBucketId(BaseRow element, Context context) {
+			try {
+				return PartitionPathUtils.generatePartitionPath(
+						computer.generatePartValues(element));
+			} catch (Exception e) {
+				throw new RuntimeException(e);
+			}
+		}
+
+		@Override
+		public SimpleVersionedSerializer<String> getSerializer() {
+			return SimpleVersionedStringSerializer.INSTANCE;
+		}
+	}
+
+	/**
+	 * Table {@link RollingPolicy}, now it is a {@link CheckpointRollingPolicy}.
+	 * Because partition commit is hard to support false.
```

Comment:
This note looks like strange,  I cannot the meaning..

Reply:
Sorry for out of date comments, should be:
```
Table {@link RollingPolicy}, it extends {@link CheckpointRollingPolicy} for bulk writers.
```